lib/iris/cube.py

```diff
@@ -2263,8 +2181,18 @@ def new_cell_measure_dims(cm_):
             if ma.count_masked(data) == 0:
                 data = data.filled()
 
+        # XXX: Slicing a single item from a masked array that is masked,
+        #      results in numpy (v1.11.1) *always* returning a MaskedConstant
+        #      with a dtype of float64, regardless of the original masked
+        #      array dtype!
+        if isinstance(data, ma.core.MaskedConstant) and \
+                data.dtype != cube_data.dtype:
+            data = ma.array(data.data, mask=data.mask, dtype=cube_data.dtype)
+
```

Comment:
Will this particular 'feature' only bite us here or will it show up in other places where we might slice single items from masked arrays?